Project/Client/LogUI.cpp

```diff
@@ -0,0 +1,311 @@
+﻿#include "pch.h"
+#include "LogUI.h"
+
+#include <Engine\CLogMgr.h>
+
+
+
+LogUI::LogUI()
+	: UI("##Log", "##LogUI")
+    , m_vectLog()
+    , m_AvailableSize(0.f, 0.f)
+    , m_iLoglvMask(0)
+    , m_iLogTimeMask(0)
+    , m_fLastScrollY(0.f)
+{
+    m_LoglvCheckBox[0] = true;
+    m_LoglvCheckBox[1] = false;
+    m_LoglvCheckBox[2] = false;
+    m_LoglvCheckBox[3] = false;
+
+    m_LogTimePrint[0] = true;
+    m_LogTimePrint[1] = false;
+    m_LogTimePrint[2] = false;
+    m_LogTimePrint[3] = false;
+
+    m_LogColor[(UINT)Log_Level::INFO] = ImVec4(0.7f, 0.7f, 0.7f, 1.0f);
+    m_LogColor[(UINT)Log_Level::WARN] = ImVec4(1.f, 1.f, 0.f, 1.0f);
+    m_LogColor[(UINT)Log_Level::ERR] = ImVec4(1.f, 0.f, 0.f, 1.0f);
+
+}
+
+LogUI::~LogUI()
+{
+}
+
+ImVec4 LogUI::SetLogColor(tLog Log)
+{
+    ImVec4 Color;
+
+    switch (Log.m_LogLv)
+    {
+    case Log_Level::INFO:
+        return Color = m_LogColor[(UINT)Log_Level::INFO];
+        break;
+    case Log_Level::WARN:
+        return Color = m_LogColor[(UINT)Log_Level::WARN];
+        break;
+    case Log_Level::ERR:
+        return Color = m_LogColor[(UINT)Log_Level::ERR];
+        break;
+    case Log_Level::END:
+        break;
+    default:
+        break;
+    }
+
+    return Color;
+}
+
+void LogUI::CheckLevelMask()
+{
+    if (m_LoglvCheckBox[0])
+    {
+        m_iLoglvMask |= 1;
+        m_iLoglvMask |= 2;
+        m_iLoglvMask |= 4;
+    }
```

Comment:
다음번엔 이런식으로 바꾸는게 더 좋을듯!

```suggestion
    if (m_LoglvCheckBox[0])
    {
        m_iLoglvMask |= LOG_LEVEL::INFO;
        m_iLoglvMask |= LOG_LEVEL::WARN;
        m_iLoglvMask |= LOG_LEVEL::ERROR;
    }
```